mriqc/cli/parser.py

```diff
@@ -520,20 +520,29 @@ def parse_args(args=None, namespace=None):
     config.workflow.analysis_level = list(analysis_level)
 
     # List of files to be run
-    bids_filters = {
-        "participant_label": config.execution.participant_label,
-        "session": config.execution.session_id,
-        "run": config.execution.run_id,
-        "task": config.execution.task_id,
-        "bids_type": config.execution.modalities,
-    }
-    config.workflow.inputs = {
-        mod: files
-        for mod, files in collect_bids_data(
-            config.execution.layout, **bids_filters
-        ).items()
-        if files
+    lc_modalities = [mod.lower() for mod in config.execution.modalities]
+
+    bids_filters = (
+        {mod: {"run": config.execution.run_id} for mod in lc_modalities}
+        if config.execution.run_id
+        else {}
+    )
```

Comment:
Do you want to go ahead and support bids filter files? https://github.com/nipreps/mriqc/issues/760

```suggestion
    bids_filters = <get base bids_filters from args>
    if config.execution.run_id:
        for mod in lc_modalities:
            bids_filters.setdefault(mod, {})["run"] = config.execution.run_id}
```

Reply:
Good idea.